pkg/services/alerting/notifiers/pagerduty.go

```diff
@@ -0,0 +1,103 @@
+package notifiers
+
+import (
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/grafana/grafana/pkg/metrics"
+	m "github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/alerting"
+)
+
+func init() {
+	alerting.RegisterNotifier("pagerduty", NewPagerdutyNotifier)
+}
+
+func NewPagerdutyNotifier(model *m.AlertNotification) (alerting.Notifier, error) {
+	key := model.Settings.Get("integrationKey").MustString()
+	if key == "" {
+		return nil, alerting.ValidationError{Reason: "Could not find integration key property in settings"}
+	}
+
+	alertingStates := make([]m.AlertStateType, 0)
```

Comment:
@arodland Storing event Id of the incident will take some more effort in PagerDuty support. Will add the feature of resolving in the next version of the support. For now when we merge this PR, Grafana will be able to Trigger an incident in PagerDuty and from there on PagerDuty can take care of the incident.